features/invitelist/impl/src/main/kotlin/io/element/android/features/invitelist/impl/components/InviteSummaryRow.kt

```diff
@@ -0,0 +1,202 @@
+            .fillMaxWidth()
+            .heightIn(min = minHeight)
+    ) {
+        DefaultInviteSummaryRow(
+            invite = invite,
+            onAcceptClicked = onAcceptClicked,
+            onDeclineClicked = onDeclineClicked,
+        )
+    }
+}
+
+@Composable
+internal fun DefaultInviteSummaryRow(
+    invite: InviteListInviteSummary,
+    onAcceptClicked: (RoomId) -> Unit = {},
+    onDeclineClicked: (RoomId) -> Unit = {},
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp, vertical = 12.dp)
+            .height(IntrinsicSize.Min),
+        verticalAlignment = Alignment.Top
+    ) {
+        Avatar(
+            invite.roomAvatarData,
+        )
+        Column(
+            modifier = Modifier
+                .padding(start = 12.dp, end = 4.dp)
+                .alignByBaseline()
+                .weight(1f)
+        ) {
+            // Name
+            Text(
+                fontSize = 16.sp,
+                fontWeight = FontWeight.SemiBold,
+                text = invite.roomName,
+                color = MaterialTheme.colorScheme.primary,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
+            )
+
+            // ID or Alias
+            invite.roomAlias?.let {
+                Text(
+                    fontSize = 14.sp,
+                    fontWeight = FontWeight.Normal,
+                    text = it,
+                    color = MaterialTheme.colorScheme.secondary,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis
+                )
+            }
+
+            // Sender
+            invite.sender?.let { sender ->
+                SenderRow(sender = sender)
+            }
+
+            // CTAs
+            Row(Modifier.padding(top = 14.dp)) {
+                OutlinedButton(
+                    content = { Text(stringResource(StringR.string.action_decline), style = ElementTextStyles.Button) },
+                    onClick = { onDeclineClicked(invite.roomId) },
+                    modifier = Modifier.weight(1f),
+                )
+
+                Spacer(modifier = Modifier.width(12.dp))
+
+                Button(
+                    content = { Text(stringResource(StringR.string.action_accept), style = ElementTextStyles.Button) },
+                    onClick = { onAcceptClicked(invite.roomId) },
+                    modifier = Modifier.weight(1f),
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun SenderRow(sender: InviteSender) {
+    Text(
+        text = buildAnnotatedString {
+            val placeholder = "$"
+            val text = stringResource(R.string.screen_invites_invited_you, placeholder)
+            val nameIndex = text.indexOf(placeholder)
+
+            // Text before the placeholder
+            append(text.take(nameIndex))
+
+            // Avatar and display name
+            appendInlineContent("avatar")
```

Comment:
I would have just used a Row here, as it seems easier. Is there any reason why you chose this instead?

Reply:
I'm not sure we can rely on the displayname always being at the start of the translated string, which is why there's the mess with the placeholder and splitting the string up (the English is currently "%1$s invited you" but I don't see why some other language with odd grammar couldn't end up as "xxx %1$s yyy" or "xxx yyy %1$s").

At that point it could be a row of something like [Text?] [Avatar] [Display name] [Text?] but the optional texts made it feel a bit fiddly, and I'm not sure how well that'd work for screen readers and the like if one short sentence is spread across many UI elements :thinking:

Reply:
Oh, I thought the avatar had to be next to the message, I didn't realise it had to be next to the user's display name as in a mention. I understand why we needed the extra complexity here then.